lib/components/app/call-taker-panel.js

```diff
@@ -0,0 +1,677 @@
+    const value = id.replace(':', '__')
+    const label = shortName
+      ? `${shortName}${longName ? ` - ${longName}` : ''}`
+      : longName
+    return {value, label}
+  }
+
+  render () {
+    const {currentQuery, modes, ModeIcon} = this.props
+    const {maxBikeDistance, maxWalkDistance, mode} = currentQuery
+    const bannedRoutes = this.getRouteList('bannedRoutes')
+    const preferredRoutes = this.getRouteList('preferredRoutes')
+    const transitModes = modes.transitModes.map(modeObj => {
+      const modeStr = modeObj.mode || modeObj
+      return {
+        id: modeStr,
+        selected: this.state.transitModes.indexOf(modeStr) !== -1,
+        text: (
+          <span>
+            <ModeIcon mode={modeStr} />
+          </span>
+        ),
+        title: modeObj.label
+      }
+    })
+    return (
+      <>
+        <div style={{alignItems: 'center', display: 'flex', justifyContent: 'space-between'}}>
+          <label style={{fontWeight: '400'}}>
+            Max walk
+            <input
+              onChange={this._setMaxWalkDistance}
+              onKeyDown={this._handleFormKeyDown}
+              step={this.getDistanceStep(maxWalkDistance)}
+              min='0'
+              style={{display: 'block', marginRight: '10px', width: '60px'}}
+              value={metersToMiles(maxWalkDistance)}
+              type='number' />
+          </label>
+          {hasBike(mode)
+            ? <label style={{fontWeight: '400'}}>
+              Max bike
+              <input
+                onChange={this._setMaxBikeDistance}
+                onKeyDown={this._handleFormKeyDown}
+                step={this.getDistanceStep(maxBikeDistance)}
+                min='0'
+                style={{display: 'block', marginRight: '10px', width: '60px'}}
+                value={metersToMiles(maxBikeDistance)}
+                type='number' />
+            </label>
+            : null
+          }
+          <StyledSubmodeSelector
+            modes={transitModes}
+            onChange={this.onSubModeChange}
+            // FIXME: Need to pass onKeyDown to children buttons in
+            // otp-ui.
+            onKeyDown={this._handleFormKeyDown}
+          />
+        </div>
+        <Select
+          value={bannedRoutes}
+          id='bannedRoutes'
+          isMulti
+          isOptionDisabled={this._isBannedRouteOptionDisabled}
+          options={this.state.routeOptions}
+          onChange={this._setBannedRoutes}
+          placeholder='Select banned routes...' />
+        <Select
+          value={preferredRoutes}
+          id='preferredRoutes'
+          isMulti
+          isOptionDisabled={this._isPreferredRouteOptionDisabled}
+          options={this.state.routeOptions}
+          onChange={this._setPreferredRoutes}
+          placeholder='Select preferred routes...' />
+      </>
+    )
+  }
+}
+
+class DateTimeOptions extends Component {
+  _setDepartArrive = evt => {
+    const {value: departArrive} = evt.target
```

Comment:
I think this might be related to an error being thrown in calltaker that is new as of some recent changes in otp-ui. I think this will help: https://github.com/opentripplanner/otp-ui/pull/193